tests/CI/run-unit-test/Dockerfile

```diff
@@ -0,0 +1,2 @@
+FROM mbedos/mbed-os-env:latest
```

Comment:
Do we even need a docker file given we just use mbed-os one and branch into the test script ? 
How different is it from invoking `tests/unittests.sh` in the workflow file ?

Reply:
Depends on whether or not you're going to expand on the docker image. I'd keep it to make it consistent - we'll use it for other CI since we need mbed tools there and it has a certain cmake version etc. don't see why we can't run it in a docker to make our life simpler